### DISCUSSION TITLE
Question: headers in the official Linux binaries.

### DISCUSSION THREAD

**Discussion by ht990332** [nodejs/help#4788] [Q&A] [answered]
Hello. This might be something obvious that I am missing but how come the official linux 64bit binaries include headers but no static or shared node library?

Thank you.

**Comment by bnoordhuis** [comment]
Node is always a single binary. You _can_ build node as a static or shared library but no precompiled images are distributed. I'm not saying it'll never happen but I don't think it's high prio or that there's much demand for it, and having too many different downloads might be confusing.

**Reply by ht990332** [reply]
Are the headers required in any case scenario with the official distribution then or can they be deleted?

**Reply by bnoordhuis** [reply, marked as answer]
Only when you want to build native modules (add-ons written in C or C++), otherwise no.
